### PR TITLE
Deploy extended-scheduler container in kube-scheduler pod

### PR DESCRIPTION
**Pull request by raja-gola**
Deploy extended-scheduler container in kube-scheduler pod

### CODE REVIEW

File: templates/kube-scheduler.yaml [hunk 1]
```diff
@@ -32,6 +32,35 @@ spec:
     volumeMounts:
     - name: kube-scheduler-policy-config
       mountPath: {{kube_scheduler_policy_config_dir}}
```
Comment (naveens) [inline templates/kube-scheduler.yaml:34]: 'if' block should be closed here. Config directory should be mounted in the scheduler container only when kube_scheduler_policy_config_dir is set.

File: templates/kube-scheduler.yaml [hunk 2]
```diff
@@ -32,6 +32,35 @@ spec:
...
+      containerPort: 9090
+    resources:
+      limits:
+        memory: {{ext_scheduler_limits_memory}}
+        cpu: {{ext_scheduler_limits_cpu}}
+      requests:
+        memory: {{ext_scheduler_requests_memory}}
+        cpu: {{ext_scheduler_requests_cpu}}
+  {% endif -%}
   volumes:
```
Comment (naveens) [inline templates/kube-scheduler.yaml]: The below volume should be created only if 'kube_scheduler_policy_config_dir' is set

File: templates/kube-scheduler.yaml (hunk 1 above)
Comment (raja-gola) [reply templates/kube-scheduler.yaml:34]: Fixed this

File: defaults/main.yml [hunk 3]
```diff
@@ -31,4 +31,13 @@ kube_scheduler_policy_config_dir: ""
 kube_scheduler_policy_config_file_name: ""
 kube_scheduler_verbosity: "3"
 
+ext_scheduler_image: ""
+ext_scheduler_label_key: ""
+ext_scheduler_log_level: ""
```
Comment (imkin) [inline defaults/main.yml]: Have a default value for this one.

File: templates/kube-scheduler.yaml [hunk 4]
```diff
@@ -32,8 +32,39 @@ spec:
...
+      containerPort: 9090
+    resources:
+      limits:
+        memory: {{ext_scheduler_limits_memory}}
+        cpu: {{ext_scheduler_limits_cpu}}
+      requests:
+        memory: {{ext_scheduler_requests_memory}}
+        cpu: {{ext_scheduler_requests_cpu}}
+  {% endif -%}
+  {% if kube_scheduler_policy_config_dir -%}
```
Comment (imkin) [inline templates/kube-scheduler.yaml:65]: What is the use of this volume mount if I do not define the "ext_scheduler_label_key" or "ext_scheduler_image" parameter.

**Review by imkin — CHANGES_REQUESTED** [review]
Is there a way to turn-off and turn-on the extended scheduler on dayN? Can you describe the workflow?

With the last scheduler workflow once we found-out that extended-scheduler was causing the problem, only by backing out the changes we could disable the extended-scheduler. Can we test the different dayN operations like update etc.,

File: templates/kube-scheduler.yaml (hunk 4 above)
Comment (raja-gola) [reply templates/kube-scheduler.yaml:65]: Volume mount is used to mount the scheduler policy config file. In the scheduler policy config file, defining extenders(by specifying URL) is not mandatory. But if you define extenders block in the policy config file, You have to set enable_ext_scheduler to True.

File: defaults/main.yml [hunk 5]
```diff
@@ -31,4 +31,6 @@ kube_scheduler_policy_config_dir: ""
 kube_scheduler_policy_config_file_name: ""
 kube_scheduler_verbosity: "3"
 
+enable_ext_scheduler: "False"
+
```
Comment (imkin) [inline defaults/main.yml]: This needs to be boolean value.

**Review by imkin — CHANGES_REQUESTED** [review]

**Comment by raja-gola** [comment]
Changed enable_ext_scheduler to boolean.

**Review by imkin — APPROVED** [review]

**Pull request merged**